test/fakeintake/server/server.go

```diff
@@ -259,42 +269,41 @@ func (fi *Server) handleDatadogRequest(w http.ResponseWriter, req *http.Request)
 
 	log.Printf("Handling Datadog %s request to %s, header %v", req.Method, req.URL.Path, req.Header)
 
-	if req.Method == http.MethodGet {
-		writeHTTPResponse(w, httpResponse{
-			statusCode: http.StatusOK,
-		})
-		return
-	}
-
-	// Datadog Agent sends a HEAD request to avoid redirect issue before sending the actual flare
-	if req.Method == http.MethodHead && req.URL.Path == "/support/flare" {
-		writeHTTPResponse(w, httpResponse{
-			statusCode: http.StatusOK,
-		})
-		return
+	switch req.Method {
+	case http.MethodPost:
+		err := fi.handleDatadogPostRequest(w, req)
+		if err == nil {
+			return
+		}
+	case http.MethodGet:
+		fallthrough
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏
TIL this keyword exists 🎉